### PR TITLE
CLDC-2020 Add saledate less than 2 weeks from today validation

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
image

ticket: https://digital.dclg.gov.uk/jira/browse/CLDC-2020

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -155,6 +155,7 @@ en:
       intermediate_rent_product_name:
         blank: "Enter name of other intermediate rent product"
       saledate:
+        later_than_14_days_after: "Sale completion date must not be later than 14 days from today’s date"
```
Comment (natdeanlewissoftwire) [inline config/locales/en.yml:158]: copy updates other than this one are not specific to this ticket, but keep the usage of "sale completion date" explicit and consistent.

File: config/locales/en.yml [hunk 2]
```diff
@@ -470,16 +471,16 @@ en:
         social_homebuy: "Social HomeBuy buyers should not have lived here before"
         rent_to_buy: "Rent to Buy buyers should not have lived here before"
       hodate:
-        must_be_before_saledate: "Practical completion or handover date must be before exchange date"
-        must_be_less_than_3_years_from_saledate: "You told us practical completion or handover date is more than 3 years before completion date"
+        must_be_before_saledate: "Practical completion or handover date must be before sale completion date"
```
Comment (natdeanlewissoftwire) [inline config/locales/en.yml:471]: this one was actually an error that I saw as part of this ticket, rest are just cosmetic changes

File: app/models/validations/sales/setup_validations.rb [hunk 3]
```diff
@@ -2,14 +2,22 @@ module Validations::Sales::SetupValidations
...
 
     unless record.saledate.between?(active_collection_start_date, current_collection_end_date)
       record.errors.add :saledate, saledate_validation_error_message
     end
   end
 
+  def validate_saledate_two_weeks(record)
+    return unless record.saledate && date_valid?("saledate", record) && FeatureToggle.saledate_two_week_validation_enabled?
+
+    if record.saledate > Time.zone.today + 14
```
Comment (jamesrose) [inline app/models/validations/sales/setup_validations.rb]:
```suggestion
    if record.saledate > Time.zone.today + 14.days
```

Same, but a little clearer :)

**Review by jamesrose — APPROVED** [review]

File: app/models/validations/sales/setup_validations.rb (hunk 3 above)
Comment (natdeanlewissoftwire) [reply app/models/validations/sales/setup_validations.rb]: yep agree, I'll copy this back to lettings too

**Review by jamesrose — APPROVED** [review]
LGTM!

**Pull request merged**